### ISSUE TITLE
Deprecate Behaviors (not a behavior) API in v3

### ISSUE THREAD

**Issue by paulfalgout** [marionettejs/backbone.marionette#3044]
We should deprecate pretty much everything in this doc.
https://github.com/marionettejs/backbone.marionette/blob/next/docs/marionette.behaviors.md

I suggest in v3 we talk about the best ways to use behaviors in the `behavior.md` file and steer people away from the `behaviors.md`

We should look into eventually removing the behaviors lookup altogether.

Module loaders have really made the behavior lookup mostly unnecessary and we can quite easily provide a simple recipe for those who really want it.  

Similarly a behavior array pattern was introduced to simplify things which is also been made unnecessary in ES6.

In the simplest case, this is all a behavior needs.
```js
import someBehavior from 'some-behavior';

const MyView = Mn.View.extend({
  behaviors: {
    someBehavior
  }
});
```

**Comment by scott-w** [comment]
Is that meant to be:

```javascript
import someBehavior from 'some-behavior';

const MyView = Mn.View.extend({
  behaviors: {
    keyName: someBehavior
  }
});
```

Or:

```javascript
import someBehavior from 'some-behavior';

const MyView = Mn.View.extend({
  behaviors: [
    someBehavior
  ]
});
```

**Comment by paulfalgout** [comment]
Well the array notation was added to prevent having to write out a key, but in es6 with shorthand object notarion the key is implied and takes the value's variable name.  Makes the array notation unnecessary

**Comment by JSteunou** [comment]
Unnecessary in es6, but have mercy for those still in es5 ;)

**Comment by paulfalgout** [comment]
Well certainly... this'd just be deprecation for the next major.. which is probably early 2017?  Then those in es5 would just need to have to add a `keyName:` to each entry.  Probably worth removing the extra API / code in the behaviors..  We can check it out later, but I think the source gets way simpler.

**Comment by scott-w** [comment]
In terms of docs, we're doing everything in ES5 syntax anyway.

**Comment by scott-w** [comment]
Just making some minor amends to the current documentation and, if array notation is supported, we might want to make it the default in all cases. I've not found the hash key that important in applications I've made.

**Comment by paulfalgout** [comment]
The key is used to determine which behavior to load from the Behavior lookup.  And it is of course necessary if you want to pass any additional options to the behavior.

**Comment by scott-w** [comment]
Alright, I'll deprecate the behavior lookups and remove references to it in the preferred documentation, so we'll not need the key for that. I'll do some playing around with it to see what the best way forward is.

**Comment by paulfalgout** [comment]
For consistency I really think the key:value is most like all of the other related interfaces..  regions and ui for example.. The array is unusual but convienent.  FWIW the enhanced object literals in ES6 solves the same problem without any additional code or documentation. And it is available without babel to IE11, Edge, FF, Chrome, Safari 9, iOS9, and Node 4 and above.  I think it is fine to keep around the array interface, but I don't think it should be the default as it is essentially a convenience for an ES5 shortcoming that is widely implemented and will soon be the standard.

**Comment by paulfalgout** [comment]
Regarding me earlier comment:
> And it is of course necessary if you want to pass any additional options to the behavior.
This wasn't true.. you can pass in options with the behaviorClass as an element of the array.

**Comment by paulfalgout** [comment]
Oh also.. we don't really need to decide this for the v3 release.  This could wait.

**Comment by scott-w** [comment]
I agree, it's not hugely urgent though I'd like to give the docs a clean-up anyway - it's difficult to see how to actually implement and attach Behaviors. I'll just do that stuff first and then we can take a step back and see what we should do after.

**Comment by ahumphreys87** [comment]
Fixed via #3072
